hadoop/src/main/java/org/apache/carbondata/hadoop/api/CarbonInputFormat.java

```diff
@@ -568,6 +569,11 @@ private int getBlockCount(List<ExtendedBlocklet> blocklets) {
     filter = filter == null ? new DataMapFilter(carbonTable, null) : filter;
     ExplainCollector.setFilterStatement(
         filter.getExpression() == null ? "none" : filter.getExpression().getStatement());
+    if (filter.isEmpty() && filter.getExpression() != null &&
```

Comment:
why this check is required??

Reply:
Have refactored the code. This check is not required anymore. Have removed it.